src/test/java/org/opentripplanner/middleware/controllers/api/UserControllerTest.java

```diff
@@ -0,0 +1,85 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.OtpMiddlewareTest;
+import org.opentripplanner.middleware.auth.Auth0UserProfile;
+import org.opentripplanner.middleware.models.User;
+import org.opentripplanner.middleware.persistence.Persistence;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+/**
+ * Tests for the UserController class.
+ */
+public class UserControllerTest extends OtpMiddlewareTest {
+
+    /**
+     * Verify the method for extracting a user from a {@link Auth0UserProfile}.
+     */
+    @Test
+    public void testGetUserFromProfile() {
+        Auth0UserProfile profile = Auth0UserProfile.createTestAdminUser();
+        User dbUser = createUser(profile.user_id, profile.email);
+
+        UserController controller = new UserController("/api/");
+        UserController.UserFromProfileResult result = controller.getUserFromProfile(profile);
+
+        assertNotNull(result.user);
+        assertNull(result.message);
+        assertEquals(profile.user_id, result.user.auth0UserId);
+
+        // tidy up
+        Persistence.users.removeById(dbUser.id);
+    }
```

Comment:
These tests feel a little too contrived. If we're making controller tests, I would rather just send http requests to the API to verify the JSON response.

Reply:
This unit test is an additional confidence check before going all in at the HTTP request level.